File: generate_sitemap.py
import os
import xml.etree.ElementTree as ET
from datetime import datetime

# Function to traverse the directory and collect URLs
def collect_urls(directory, base_url):
    urls = []
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith(".html"):
                file_path = os.path.join(root, file)
                url = file_path.replace(directory, base_url).replace("\\", "/")
                urls.append(url)
    return urls

# Function to create the sitemap XML
def create_sitemap(urls, output_file):
    urlset = ET.Element("urlset", xmlns="http://www.sitemaps.org/schemas/sitemap/0.9")

    for url in urls:
        url_element = ET.Element("url")
        loc = ET.Element("loc")
        loc.text = url
        lastmod = ET.Element("lastmod")
        lastmod.text = datetime.now().strftime("%Y-%m-%d")

        url_element.append(loc)
        url_element.append(lastmod)
        urlset.append(url_element)

    tree = ET.ElementTree(urlset)
    tree.write(output_file, encoding="utf-8", xml_declaration=True)

# Define the main function
def main():
    # Define your site's base URL and directory to scan
    base_url = "https://monvvo.com"
    directory_to_scan = "C:\Users\Tyler\Documents\GitHub\Monvvo-App"  # Update this to your site's root directory
    output_file = "sitemap.xml"

    # Collect URLs from the directory
    urls = collect_urls(directory_to_scan, base_url)

    # Create the sitemap
    create_sitemap(urls, output_file)

    print(f"Sitemap has been generated and saved to {output_file}")

if __name__ == "__main__":
    main()

File: generate-sitemap.js
// generate-sitemap.js
const fs = require('fs');
const path = require('path');

const baseDir = path.join(__dirname, 'public');
const sitemapPath = path.join(baseDir, 'sitemap.xml');
const baseUrl = 'https://www.monvvo.com'; // Replace with your actual domain

const pages = [
  { dir: 'Loan', file: 'Auto-Loan-Calculator.html', priority: 0.8 },
  { dir: 'Loan', file: 'Loan-Amortization-Calculator.html', priority: 0.8 },
  { dir: 'Loan', file: 'Personal-Loan-Calculator.html', priority: 0.8 },
  { dir: 'Loan', file: 'Student-Loan-Calculator.html', priority: 0.8 },
  { dir: 'Investment', file: 'Compound-Interest-Calculator.html', priority: 0.7 },
  { dir: 'Investment', file: 'Investment-Growth-Calculator.html', priority: 0.7 },
  { dir: 'Investment', file: 'Retirement-Savings-Calculator.html', priority: 0.7 },
  { dir: 'Mortgage', file: 'Down-Payment-Calculator.html', priority: 0.9 },
  { dir: 'Mortgage', file: 'Home-Equity-Calculator.html', priority: 0.9 },
  { dir: 'Mortgage', file: 'Mortgage-Payoff-Calculator.html', priority: 0.9 }
];

function generateSitemap() {
  try {
    const urls = pages.map(page => {
      const lastmod = new Date().toISOString().split('T')[0]; // current date in YYYY-MM-DD format
      return `
        <url>
          <loc>${baseUrl}/${page.dir}/${page.file}</loc>
          <lastmod>${lastmod}</lastmod>
          <changefreq>monthly</changefreq>
          <priority>${page.priority}</priority>
        </url>`;
    });

    const sitemapContent = `
      <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
        ${urls.join('\n')}
      </urlset>
    `;

    fs.writeFileSync(sitemapPath, sitemapContent.trim());
    console.log('Sitemap generated successfully.');
  } catch (error) {
    console.error('Error generating sitemap:', error);
  }
}

generateSitemap();
